### PR TITLE
Rework startup & configuration

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Rework startup & configuration

This brings better behaviour when installed on a desktop system:

  o daemon defaults to false
  o can be run on an X11 desktop (including, with devmode, as root)

This makes testing kiosk snaps a lot easier.

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
There's an amd64 build: `--channel=edge/rework`

File: snap/hooks/post-refresh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+set -e
+
+display_layout=$(snapctl get display-layout)
```
Comment (wmww) [inline snap/hooks/post-refresh:4]: Does all this code have to be duplicated between install and post-refresh? Wouldn't it make sense to have a helper script run by each?

**Review by wmww — COMMENTED** [review]
Looks good, except for the logic duplication in the hooks

File: snap/hooks/post-refresh (hunk 1 above)
Comment (AlanGriffiths) [reply snap/hooks/post-refresh:4]: Yes, they did end up _almost_ the same. I'm not convinced de-duplication is enough motivation for adding a script for the common bits.

**Review by wmww — APPROVED** [review]
Looks good
bors r+

**Pull request merged**